Java SDK 1.0.8 is noisy on shutdown

## Expected Behavior

Graceful worker shutdown should result in no errors in application logs.

## Actual Behavior

As of Java SDK 1.0.8 workers tend to log some errors like

```
java.lang.InterruptedException (Most recent call first)
    at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted (ClientCalls.java:749)
    at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain (ClientCalls.java:732)
    at io.grpc.stub.ClientCalls.blockingUnaryCall (ClientCalls.java:149)
    at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue (WorkflowServiceGrpc.java:2639)
    at io.temporal.internal.worker.WorkflowPollTask.poll (WorkflowPollTask.java:81)
    at io.temporal.internal.worker.WorkflowPollTask.poll (WorkflowPollTask.java:37)
    at io.temporal.internal.worker.Poller$PollExecutionTask.run (Poller.java:283)
    at io.temporal.internal.worker.Poller$PollLoopTask.run (Poller.java:242)
    at java.util.concurrent.ThreadPoolExecutor.runWorker ([unknown]:-1)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run ([unknown]:-1)
    at java.lang.Thread.run ([unknown]:-1)
io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted (Most recent call first)
    at io.grpc.stub.ClientCalls.toStatusRuntimeException (ClientCalls.java:262)
    at io.grpc.stub.ClientCalls.getUnchecked (ClientCalls.java:243)
    at io.grpc.stub.ClientCalls.blockingUnaryCall (ClientCalls.java:156)
    at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue (WorkflowServiceGrpc.java:2639)
    at io.temporal.internal.worker.WorkflowPollTask.poll (WorkflowPollTask.java:81)
    at io.temporal.internal.worker.WorkflowPollTask.poll (WorkflowPollTask.java:37)
    at io.temporal.internal.worker.Poller$PollExecutionTask.run (Poller.java:283)
    at io.temporal.internal.worker.Poller$PollLoopTask.run (Poller.java:242)
    at java.util.concurrent.ThreadPoolExecutor.runWorker ([unknown]:-1)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run ([unknown]:-1)
    at java.lang.Thread.run ([unknown]:-1) 
```

## Steps to Reproduce the Problem

1. Start JVM worker.
2. Send it a signal for graceful termination.
3. Watch application logs.

The issue could be reproduced with any sample from the [`samples-java`](https://github.com/temporalio/samples-java/) repository. E.g. with [`HelloActivity`](https://github.com/temporalio/samples-java/blob/84e9f2cf986742c7a0cc56d57a8f7299f18ae0ab/src/main/java/io/temporal/samples/hello/HelloActivity.java):

<details>
<summary>Add factory shutdown code to the example</summary>

```diff
Index: src/main/java/io/temporal/samples/hello/HelloActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/io/temporal/samples/hello/HelloActivity.java b/src/main/java/io/temporal/samples/hello/HelloActivity.java
--- a/src/main/java/io/temporal/samples/hello/HelloActivity.java	(revision 7021fe356c72489c41048774b329054422f42144)
+++ b/src/main/java/io/temporal/samples/hello/HelloActivity.java	(date 1622552592206)
@@ -31,6 +31,7 @@
 import io.temporal.workflow.WorkflowInterface;
 import io.temporal.workflow.WorkflowMethod;
 import java.time.Duration;
+import java.util.concurrent.TimeUnit;
 
 /**
  * Sample Temporal workflow that executes a single workflow activity.
@@ -183,6 +184,10 @@
 
     // Display workflow execution results
     System.out.println(greeting);
+
+    factory.shutdown();
+    factory.awaitTermination(1, TimeUnit.MINUTES);
+
     System.exit(0);
   }
 }    
```
</details>

<details>
<summary>Temporal SDK 1.0.7 logs</summary>

```
16:03:18.562 [main] INFO  i.t.s.WorkflowServiceStubsImpl - Created GRPC client for channel: ManagedChannelOrphanWrapper{delegate=ManagedChannelImpl{logId=1, target=127.0.0.1:7233}}
16:03:20.878 [main] INFO  io.temporal.internal.worker.Poller - start(): Poller{options=PollerOptions{maximumPollRateIntervalMilliseconds=1000, maximumPollRatePerSecond=0.0, pollBackoffCoefficient=2.0, pollBackoffInitialInterval=PT0.1S, pollBackoffMaximumInterval=PT1M, pollThreadCount=2, pollThreadNamePrefix='Workflow Poller taskQueue="HelloActivityTaskQueue", namespace="default"'}, identity=89225@KV0200-Mykola-Rybak}
16:03:20.879 [main] INFO  io.temporal.internal.worker.Poller - start(): Poller{options=PollerOptions{maximumPollRateIntervalMilliseconds=1000, maximumPollRatePerSecond=0.0, pollBackoffCoefficient=2.0, pollBackoffInitialInterval=PT0.1S, pollBackoffMaximumInterval=PT1M, pollThreadCount=1, pollThreadNamePrefix='Local Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default"'}, identity=89225@KV0200-Mykola-Rybak}
16:03:20.881 [main] INFO  io.temporal.internal.worker.Poller - start(): Poller{options=PollerOptions{maximumPollRateIntervalMilliseconds=1000, maximumPollRatePerSecond=0.0, pollBackoffCoefficient=2.0, pollBackoffInitialInterval=PT0.1S, pollBackoffMaximumInterval=PT1M, pollThreadCount=5, pollThreadNamePrefix='Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default"'}, identity=89225@KV0200-Mykola-Rybak}
16:03:20.882 [main] INFO  io.temporal.internal.worker.Poller - start(): Poller{options=PollerOptions{maximumPollRateIntervalMilliseconds=1000, maximumPollRatePerSecond=0.0, pollBackoffCoefficient=2.0, pollBackoffInitialInterval=PT0.1S, pollBackoffMaximumInterval=PT1M, pollThreadCount=5, pollThreadNamePrefix='Host Local Workflow Poller'}, identity=b8f3f995-bd96-4074-bd05-ec5f7432d50a}
Hello World!
16:03:21.382 [main] INFO  io.temporal.worker.WorkerFactory - shutdown
16:03:21.382 [main] INFO  io.temporal.internal.worker.Poller - shutdown
16:03:21.383 [Host Local Workflow Poller: 1] INFO  io.temporal.internal.worker.Poller - poll loop done
16:03:21.383 [Host Local Workflow Poller: 3] INFO  io.temporal.internal.worker.Poller - poll loop done
16:03:21.383 [Host Local Workflow Poller: 2] INFO  io.temporal.internal.worker.Poller - poll loop done
16:03:21.383 [Host Local Workflow Poller: 5] INFO  io.temporal.internal.worker.Poller - poll loop done
16:03:21.383 [Host Local Workflow Poller: 4] INFO  io.temporal.internal.worker.Poller - poll loop done
16:03:21.384 [main] INFO  io.temporal.internal.worker.Poller - shutdown
16:03:21.384 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 1] INFO  io.temporal.internal.worker.Poller - poll loop done
16:03:21.385 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 5] INFO  io.temporal.internal.worker.Poller - poll loop done
16:03:21.385 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 4] INFO  io.temporal.internal.worker.Poller - poll loop done
16:03:21.385 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 3] INFO  io.temporal.internal.worker.Poller - poll loop done
16:03:21.385 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 2] INFO  io.temporal.internal.worker.Poller - poll loop done
16:03:21.385 [main] INFO  io.temporal.internal.worker.Poller - shutdown
16:03:21.385 [Local Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 1] INFO  io.temporal.internal.worker.Poller - poll loop done
16:03:21.385 [main] INFO  io.temporal.internal.worker.Poller - shutdown
16:03:21.385 [Workflow Poller taskQueue="HelloActivityTaskQueue", namespace="default": 1] INFO  io.temporal.internal.worker.Poller - poll loop done
16:03:21.385 [Workflow Poller taskQueue="HelloActivityTaskQueue", namespace="default": 2] INFO  io.temporal.internal.worker.Poller - poll loop done
16:03:21.386 [main] INFO  io.temporal.worker.WorkerFactory - awaitTermination begin
16:03:21.386 [main] INFO  io.temporal.worker.WorkerFactory - awaitTermination done
```
</details>

<details>
<summary>Temporal SDK 1.0.8 logs</summary>

```
16:03:49.630 [main] INFO  i.t.s.WorkflowServiceStubsImpl - Created GRPC client for channel: ManagedChannelOrphanWrapper{delegate=ManagedChannelImpl{logId=1, target=127.0.0.1:7233}}
16:03:51.986 [main] INFO  io.temporal.internal.worker.Poller - start(): Poller{options=PollerOptions{maximumPollRateIntervalMilliseconds=1000, maximumPollRatePerSecond=0.0, pollBackoffCoefficient=2.0, pollBackoffInitialInterval=PT0.1S, pollBackoffMaximumInterval=PT1M, pollThreadCount=2, pollThreadNamePrefix='Workflow Poller taskQueue="HelloActivityTaskQueue", namespace="default"'}, identity=89263@KV0200-Mykola-Rybak}
16:03:51.988 [main] INFO  io.temporal.internal.worker.Poller - start(): Poller{options=PollerOptions{maximumPollRateIntervalMilliseconds=1000, maximumPollRatePerSecond=0.0, pollBackoffCoefficient=2.0, pollBackoffInitialInterval=PT0.1S, pollBackoffMaximumInterval=PT1M, pollThreadCount=1, pollThreadNamePrefix='Local Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default"'}, identity=89263@KV0200-Mykola-Rybak}
16:03:51.989 [main] INFO  io.temporal.internal.worker.Poller - start(): Poller{options=PollerOptions{maximumPollRateIntervalMilliseconds=1000, maximumPollRatePerSecond=0.0, pollBackoffCoefficient=2.0, pollBackoffInitialInterval=PT0.1S, pollBackoffMaximumInterval=PT1M, pollThreadCount=5, pollThreadNamePrefix='Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default"'}, identity=89263@KV0200-Mykola-Rybak}
16:03:51.990 [main] INFO  io.temporal.internal.worker.Poller - start(): Poller{options=PollerOptions{maximumPollRateIntervalMilliseconds=1000, maximumPollRatePerSecond=0.0, pollBackoffCoefficient=2.0, pollBackoffInitialInterval=PT0.1S, pollBackoffMaximumInterval=PT1M, pollThreadCount=5, pollThreadNamePrefix='Host Local Workflow Poller'}, identity=a9903edb-c525-4369-bf3c-926eb856c7d2}
Hello World!
16:03:52.543 [main] INFO  io.temporal.worker.WorkerFactory - shutdown
16:03:52.543 [main] INFO  io.temporal.internal.worker.Poller - shutdown
16:03:53.052 [Host Local Workflow Poller: 5] ERROR io.temporal.internal.worker.Poller - Failure in thread Host Local Workflow Poller: 5
io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue(WorkflowServiceGrpc.java:2639)
	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:81)
	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:37)
	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException: null
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
	... 8 common frames omitted
16:03:53.052 [Host Local Workflow Poller: 1] ERROR io.temporal.internal.worker.Poller - Failure in thread Host Local Workflow Poller: 1
io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue(WorkflowServiceGrpc.java:2639)
	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:81)
	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:37)
	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException: null
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
	... 8 common frames omitted
16:03:53.052 [Host Local Workflow Poller: 4] ERROR io.temporal.internal.worker.Poller - Failure in thread Host Local Workflow Poller: 4
io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue(WorkflowServiceGrpc.java:2639)
	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:81)
	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:37)
	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException: null
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
	... 8 common frames omitted
16:03:53.053 [Host Local Workflow Poller: 5] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
16:03:53.052 [Host Local Workflow Poller: 3] ERROR io.temporal.internal.worker.Poller - Failure in thread Host Local Workflow Poller: 3
io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue(WorkflowServiceGrpc.java:2639)
	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:81)
	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:37)
	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException: null
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
	... 8 common frames omitted
16:03:53.053 [Host Local Workflow Poller: 4] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
16:03:53.052 [Host Local Workflow Poller: 2] ERROR io.temporal.internal.worker.Poller - Failure in thread Host Local Workflow Poller: 2
io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue(WorkflowServiceGrpc.java:2639)
	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:81)
	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:37)
	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException: null
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
	... 8 common frames omitted
16:03:53.053 [Host Local Workflow Poller: 1] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
16:03:53.053 [Host Local Workflow Poller: 2] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
16:03:53.053 [Host Local Workflow Poller: 3] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
16:03:53.055 [main] INFO  io.temporal.internal.worker.Poller - shutdown
16:03:53.558 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 1] ERROR io.temporal.internal.worker.Poller - Failure in thread Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 1
io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollActivityTaskQueue(WorkflowServiceGrpc.java:2683)
	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:99)
	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:39)
	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException: null
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
	... 8 common frames omitted
16:03:53.558 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 2] ERROR io.temporal.internal.worker.Poller - Failure in thread Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 2
io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollActivityTaskQueue(WorkflowServiceGrpc.java:2683)
	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:99)
	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:39)
	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException: null
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
	... 8 common frames omitted
16:03:53.558 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 1] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
16:03:53.558 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 2] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
16:03:53.558 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 4] ERROR io.temporal.internal.worker.Poller - Failure in thread Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 4
io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollActivityTaskQueue(WorkflowServiceGrpc.java:2683)
	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:99)
	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:39)
	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException: null
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
	... 8 common frames omitted
16:03:53.558 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 5] ERROR io.temporal.internal.worker.Poller - Failure in thread Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 5
io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollActivityTaskQueue(WorkflowServiceGrpc.java:2683)
	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:99)
	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:39)
	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException: null
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
	... 8 common frames omitted
16:03:53.558 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 5] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
16:03:53.558 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 4] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
16:03:53.559 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 3] ERROR io.temporal.internal.worker.Poller - Failure in thread Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 3
io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollActivityTaskQueue(WorkflowServiceGrpc.java:2683)
	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:99)
	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:39)
	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException: null
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
	... 8 common frames omitted
16:03:53.559 [Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 3] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
16:03:53.559 [main] INFO  io.temporal.internal.worker.Poller - shutdown
16:03:54.062 [Local Activity Poller taskQueue="HelloActivityTaskQueue", namespace="default": 1] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
16:03:54.062 [main] INFO  io.temporal.internal.worker.Poller - shutdown
16:03:54.565 [Workflow Poller taskQueue="HelloActivityTaskQueue", namespace="default": 1] ERROR io.temporal.internal.worker.Poller - Failure in thread Workflow Poller taskQueue="HelloActivityTaskQueue", namespace="default": 1
io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue(WorkflowServiceGrpc.java:2639)
	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:81)
	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:37)
	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException: null
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
	... 8 common frames omitted
16:03:54.566 [Workflow Poller taskQueue="HelloActivityTaskQueue", namespace="default": 1] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
16:03:54.566 [Workflow Poller taskQueue="HelloActivityTaskQueue", namespace="default": 2] ERROR io.temporal.internal.worker.Poller - Failure in thread Workflow Poller taskQueue="HelloActivityTaskQueue", namespace="default": 2
io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue(WorkflowServiceGrpc.java:2639)
	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:81)
	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:37)
	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException: null
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
	... 8 common frames omitted
16:03:54.566 [Workflow Poller taskQueue="HelloActivityTaskQueue", namespace="default": 2] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
16:03:54.566 [main] INFO  io.temporal.worker.WorkerFactory - awaitTermination begin
16:03:54.567 [main] INFO  io.temporal.worker.WorkerFactory - awaitTermination done
```
</details>

**Update**: the same issue is also reproduced by Java SDK builds itself. Here's a part of the build log with `USE_DOCKER_SERVICE=true`:

<details>
<summary>Java SDK build log</summary>

```
io.temporal.workflow.GenericParametersWorkflowTest > testGenericParametersWorkflow STANDARD_OUT
    07:43:48.111 [Test worker] INFO  i.t.s.WorkflowServiceStubsImpl - Created GRPC client for channel: ManagedChannelOrphanWrapper{delegate=ManagedChannelImpl{logId=4, target=temporal:7233}}
    07:43:48.862 [Time-limited test] INFO  io.temporal.internal.worker.Poller - start(): Poller{options=PollerOptions{maximumPollRateIntervalMilliseconds=1000, maximumPollRatePerSecond=0.0, pollBackoffCoefficient=2.0, pollBackoffInitialInterval=PT0.1S, pollBackoffMaximumInterval=PT1M, pollThreadCount=2, pollThreadNamePrefix='Workflow Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest"'}, identity=3895@e0402adf52a3}
    07:43:48.881 [Time-limited test] INFO  io.temporal.internal.worker.Poller - start(): Poller{options=PollerOptions{maximumPollRateIntervalMilliseconds=1000, maximumPollRatePerSecond=0.0, pollBackoffCoefficient=2.0, pollBackoffInitialInterval=PT0.1S, pollBackoffMaximumInterval=PT1M, pollThreadCount=1, pollThreadNamePrefix='Local Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest"'}, identity=3895@e0402adf52a3}
    07:43:48.896 [Time-limited test] INFO  io.temporal.internal.worker.Poller - start(): Poller{options=PollerOptions{maximumPollRateIntervalMilliseconds=1000, maximumPollRatePerSecond=0.0, pollBackoffCoefficient=2.0, pollBackoffInitialInterval=PT0.1S, pollBackoffMaximumInterval=PT1M, pollThreadCount=5, pollThreadNamePrefix='Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest"'}, identity=3895@e0402adf52a3}
    07:43:48.902 [Time-limited test] INFO  io.temporal.internal.worker.Poller - start(): Poller{options=PollerOptions{maximumPollRateIntervalMilliseconds=1000, maximumPollRatePerSecond=0.0, pollBackoffCoefficient=2.0, pollBackoffInitialInterval=PT0.1S, pollBackoffMaximumInterval=PT1M, pollThreadCount=5, pollThreadNamePrefix='Host Local Workflow Poller'}, identity=bbd3b7ae-1225-4ce3-bdfe-fd3bdc6b49d2}
    07:43:52.956 [Time-limited test] INFO  io.temporal.worker.WorkerFactory - shutdownNow
    07:43:52.957 [Time-limited test] INFO  io.temporal.internal.worker.Poller - shutdownNow poller=Host Local Workflow Poller
    07:43:53.474 [Host Local Workflow Poller: 4] ERROR io.temporal.internal.worker.Poller - Failure in thread Host Local Workflow Poller: 4
    io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
    	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
    	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
    	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue(WorkflowServiceGrpc.java:2639)
    	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:81)
    	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:37)
    	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
    	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    	at java.lang.Thread.run(Thread.java:748)
    Caused by: java.lang.InterruptedException: null
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
    	... 8 common frames omitted
    07:43:53.474 [Host Local Workflow Poller: 3] ERROR io.temporal.internal.worker.Poller - Failure in thread Host Local Workflow Poller: 3
    io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
    	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
    	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
    	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue(WorkflowServiceGrpc.java:2639)
    	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:81)
    	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:37)
    	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
    	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    	at java.lang.Thread.run(Thread.java:748)
    Caused by: java.lang.InterruptedException: null
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
    	... 8 common frames omitted
    07:43:53.475 [Host Local Workflow Poller: 2] ERROR io.temporal.internal.worker.Poller - Failure in thread Host Local Workflow Poller: 2
    io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
    	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
    	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
    	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue(WorkflowServiceGrpc.java:2639)
    	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:81)
    	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:37)
    	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
    	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    	at java.lang.Thread.run(Thread.java:748)
    Caused by: java.lang.InterruptedException: null
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
    	... 8 common frames omitted
    07:43:53.474 [Host Local Workflow Poller: 5] ERROR io.temporal.internal.worker.Poller - Failure in thread Host Local Workflow Poller: 5
    io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
    	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
    	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
    	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue(WorkflowServiceGrpc.java:2639)
    	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:81)
    	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:37)
    	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
    	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    	at java.lang.Thread.run(Thread.java:748)
    Caused by: java.lang.InterruptedException: null
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
    	... 8 common frames omitted
    07:43:53.475 [Host Local Workflow Poller: 4] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
    07:43:53.478 [Host Local Workflow Poller: 5] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
    07:43:53.476 [Host Local Workflow Poller: 3] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
    07:43:53.477 [Host Local Workflow Poller: 2] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
    07:43:53.481 [Host Local Workflow Poller: 1] ERROR io.temporal.internal.worker.Poller - Failure in thread Host Local Workflow Poller: 1
    io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
    	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
    	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
    	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue(WorkflowServiceGrpc.java:2639)
    	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:81)
    	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:37)
    	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
    	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    	at java.lang.Thread.run(Thread.java:748)
    Caused by: java.lang.InterruptedException: null
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
    	... 8 common frames omitted
    07:43:53.482 [Host Local Workflow Poller: 1] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
    07:43:53.487 [Time-limited test] INFO  io.temporal.internal.worker.Poller - shutdownNow poller=Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest"
    07:43:53.988 [Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 3] ERROR io.temporal.internal.worker.Poller - Failure in thread Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 3
    io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
    	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
    	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
    	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollActivityTaskQueue(WorkflowServiceGrpc.java:2683)
    	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:99)
    	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:39)
    	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
    	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    	at java.lang.Thread.run(Thread.java:748)
    Caused by: java.lang.InterruptedException: null
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
    	... 8 common frames omitted
    07:43:53.989 [Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 2] ERROR io.temporal.internal.worker.Poller - Failure in thread Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 2
    io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
    	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
    	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
    	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollActivityTaskQueue(WorkflowServiceGrpc.java:2683)
    	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:99)
    	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:39)
    	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
    	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    	at java.lang.Thread.run(Thread.java:748)
    Caused by: java.lang.InterruptedException: null
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
    	... 8 common frames omitted
    07:43:53.990 [Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 2] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
    07:43:53.990 [Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 4] ERROR io.temporal.internal.worker.Poller - Failure in thread Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 4
    io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
    	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
    	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
    	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollActivityTaskQueue(WorkflowServiceGrpc.java:2683)
    	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:99)
    	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:39)
    	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
    	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    	at java.lang.Thread.run(Thread.java:748)
    Caused by: java.lang.InterruptedException: null
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
    	... 8 common frames omitted
    07:43:53.991 [Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 5] ERROR io.temporal.internal.worker.Poller - Failure in thread Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 5
    io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
    	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
    	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
    	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollActivityTaskQueue(WorkflowServiceGrpc.java:2683)
    	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:99)
    	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:39)
    	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
    	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    	at java.lang.Thread.run(Thread.java:748)
    Caused by: java.lang.InterruptedException: null
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
    	... 8 common frames omitted
    07:43:53.993 [Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 5] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
    07:43:53.990 [Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 1] ERROR io.temporal.internal.worker.Poller - Failure in thread Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 1
    io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
    	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
    	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
    	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollActivityTaskQueue(WorkflowServiceGrpc.java:2683)
    	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:99)
    	at io.temporal.internal.worker.ActivityPollTask.poll(ActivityPollTask.java:39)
    	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
    	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    	at java.lang.Thread.run(Thread.java:748)
    Caused by: java.lang.InterruptedException: null
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
    	... 8 common frames omitted
    07:43:53.989 [Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 3] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
    07:43:53.994 [Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 1] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
    07:43:53.999 [Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 4] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
    07:43:54.003 [Time-limited test] INFO  io.temporal.internal.worker.Poller - shutdownNow poller=Local Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest"

io.temporal.workflow.MetricsTest > testWorkflowMetrics STANDARD_OUT
    07:43:54.243 [Test worker] INFO  i.t.s.WorkflowServiceStubsImpl - Created GRPC client for channel: ManagedChannelOrphanWrapper{delegate=ManagedChannelImpl{logId=1, target=directaddress:///1e7752aa-0225-4ff2-be30-240124bc806e}}
    07:43:54.493 [Test worker] INFO  i.t.s.WorkflowServiceStubsImpl - Created GRPC client for channel: ManagedChannelOrphanWrapper{delegate=ManagedChannelImpl{logId=5, target=directaddress:///133b8efe-30a7-484a-95c7-6bd988c95170}}

io.temporal.workflow.GenericParametersWorkflowTest > testGenericParametersWorkflow STANDARD_OUT
    07:43:54.504 [Local Activity Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 1] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
    07:43:54.506 [Time-limited test] INFO  io.temporal.internal.worker.Poller - shutdownNow poller=Workflow Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest"
    07:43:55.008 [Workflow Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 2] ERROR io.temporal.internal.worker.Poller - Failure in thread Workflow Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 2
    io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
    	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
    	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
    	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue(WorkflowServiceGrpc.java:2639)
    	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:81)
    	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:37)
    	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
    	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    	at java.lang.Thread.run(Thread.java:748)
    Caused by: java.lang.InterruptedException: null
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
    	... 8 common frames omitted
    07:43:55.010 [Workflow Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 2] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
    07:43:55.014 [Workflow Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 1] ERROR io.temporal.internal.worker.Poller - Failure in thread Workflow Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 1
    io.grpc.StatusRuntimeException: CANCELLED: Thread interrupted
    	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
    	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
    	at io.temporal.api.workflowservice.v1.WorkflowServiceGrpc$WorkflowServiceBlockingStub.pollWorkflowTaskQueue(WorkflowServiceGrpc.java:2639)
    	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:81)
    	at io.temporal.internal.worker.WorkflowPollTask.poll(WorkflowPollTask.java:37)
    	at io.temporal.internal.worker.Poller$PollExecutionTask.run(Poller.java:283)
    	at io.temporal.internal.worker.Poller$PollLoopTask.run(Poller.java:242)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    	at java.lang.Thread.run(Thread.java:748)
    Caused by: java.lang.InterruptedException: null
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.throwIfInterrupted(ClientCalls.java:749)
    	at io.grpc.stub.ClientCalls$ThreadlessExecutor.waitAndDrain(ClientCalls.java:732)
    	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:149)
    	... 8 common frames omitted
    07:43:55.017 [Workflow Poller taskQueue="WorkflowTest-testGenericParametersWorkflow-867aa9ec-3479-478a-bf63-5fe8a3f87253", namespace="UnitTest": 1] INFO  io.temporal.internal.worker.Poller - poll loop is terminated
    07:43:55.019 [Time-limited test] INFO  io.temporal.worker.WorkerFactory - awaitTermination begin
    07:43:55.024 [Time-limited test] INFO  io.temporal.worker.WorkerFactory - awaitTermination done
    07:43:55.026 [Time-limited test] INFO  i.t.i.t.TestWorkflowService - Shutting down GRPC server
    07:43:55.061 [Time-limited test] INFO  i.t.s.WorkflowServiceStubsImpl - shutdown

io.temporal.workflow.GenericParametersWorkflowTest > testGenericParametersWorkflow PASSED
```
</details>

## Specifications

  - Version: Java SDK 1.0.8
  - Platform: JDK 11.0.11

--- a/temporal-sdk/src/main/java/io/temporal/internal/worker/Poller.java
+++ b/temporal-sdk/src/main/java/io/temporal/internal/worker/Poller.java
@@ -76,6 +76,12 @@ public final class Poller<T> implements SuspendableWorker {
             log.warn("Failure in thread " + t.getName(), e);
             return;
           }
+          if (te.getCause() instanceof InterruptedException) {
+            if (log.isDebugEnabled()) {
+              log.debug("Failure in thread " + t.getName(), e);
+            }
+            return;
+          }
         }
         log.error("Failure in thread " + t.getName(), e);
       };

--- a/temporal-sdk/src/main/java/io/temporal/internal/worker/Poller.java
+++ b/temporal-sdk/src/main/java/io/temporal/internal/worker/Poller.java
@@ -77,9 +77,7 @@ public final class Poller<T> implements SuspendableWorker {
             return;
           }
           if (te.getCause() instanceof InterruptedException) {
-            if (log.isDebugEnabled()) {
-              log.debug("Failure in thread " + t.getName(), e);
-            }
+            log.debug("Failure in thread " + t.getName(), e);
             return;
           }
         }

--- a/temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java
+++ b/temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java
@@ -348,9 +348,8 @@ public class TestWorkflowRule implements TestRule {
   }
 
   /**
-   * Returns the default worker created for each test method.
-   * This worker listens to the default task queue which is obtainable
-   * via the {@link #getTaskQueue()} method.
+   * Returns the default worker created for each test method. This worker listens to the default
+   * task queue which is obtainable via the {@link #getTaskQueue()} method.
    */
   public Worker getWorker() {
     return testEnvironment.getWorkerFactory().getWorker(getTaskQueue());

